### PR TITLE
make sure quotes in the labels are escaped

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/4630

### CODE REVIEW

**Comment by bparees** [comment]
@liggitt @csrwng ptal.

**Comment by csrwng** [comment]
LGTM

File: pkg/build/builder/docker.go [hunk 1]
```diff
@@ -276,10 +275,10 @@ func appendMetadata(inst MetaInstruction, fileData string, envVars map[string]st
 	first := true
 	for k, v := range envVars {
 		if first {
-			fileData += fmt.Sprintf("%s %s=\"%s\"", inst, k, v)
+			fileData += fmt.Sprintf("%s %s=%+q", inst, k, v)
 			first = false
 		} else {
-			fileData += fmt.Sprintf(" \\\n\t%s=\"%s\"", k, v)
+			fileData += fmt.Sprintf(" \\\n\t%s=%+q", k, v)
```
Comment (liggitt) [inline pkg/build/builder/docker.go]: lgtm, as a tip, you can use `` and put raw unescaped stuff in your string, like this:
```
			fileData += fmt.Sprintf(` \
	%s=%+q`, k, v)
```

**Comment by bparees** [comment]
[merge]

**Pull request merged**